src/modules/Game.class.js

```diff
@@ -1,68 +1,216 @@
+    if (this.status === Game.gameStatus.playing) {
+      for (let row = 0; row < 4; row++) {
+        let newRow = this.state[row].filter((val) => val !== 0);
+
+        for (let col = newRow.length - 1; col > 0; col--) {
+          if (newRow[col] === newRow[col - 1]) {
+            newRow[col] *= 2;
+            this.score += newRow[col];
+            newRow[col - 1] = 0;
+          }
+        }
+        newRow = newRow.filter((val) => val !== 0);
+
+        while (newRow.length < 4) {
+          newRow.unshift(0);
+        }
+        this.state[row] = newRow;
+      }
+
+      this.getRandomCell();
+      this.checkStatus();
+    }
+  }
+  moveUp() {
+    if (this.status === Game.gameStatus.playing) {
+      for (let col = 0; col < 4; col++) {
+        let newCol = [];
+
+        for (let row = 0; row < 4; row++) {
+          if (this.state[row][col] !== 0) {
+            newCol.push(this.state[row][col]);
+          }
+        }
+
+        for (let row = 0; row < newCol.length - 1; row++) {
+          if (newCol[row] === newCol[row + 1]) {
+            newCol[row] *= 2;
+            this.score += newCol[row];
+            newCol[row + 1] = 0;
+          }
+        }
+        newCol = newCol.filter((val) => val !== 0);
+
+        while (newCol.length < 4) {
+          newCol.push(0);
+        }
+
+        for (let row = 0; row < 4; row++) {
+          this.state[row][col] = newCol[row];
+        }
+      }
+
+      this.getRandomCell();
+      this.checkStatus();
+    }
   }
 
-  moveLeft() {}
-  moveRight() {}
-  moveUp() {}
-  moveDown() {}
-
-  /**
-   * @returns {number}
-   */
-  getScore() {}
-
-  /**
-   * @returns {number[][]}
-   */
-  getState() {}
-
-  /**
-   * Returns the current game status.
-   *
-   * @returns {string} One of: 'idle', 'playing', 'win', 'lose'
-   *
-   * `idle` - the game has not started yet (the initial state);
-   * `playing` - the game is in progress;
-   * `win` - the game is won;
-   * `lose` - the game is lost
-   */
-  getStatus() {}
-
-  /**
-   * Starts the game.
-   */
-  start() {}
-
-  /**
-   * Resets the game.
-   */
-  restart() {}
-
-  // Add your own methods here
+  moveDown() {
+    if (this.status === Game.gameStatus.playing) {
+      for (let col = 0; col < 4; col++) {
+        let newCol = [];
+
+        for (let row = 0; row < 4; row++) {
+          if (this.state[row][col] !== 0) {
+            newCol.push(this.state[row][col]);
+          }
+        }
+
+        for (let row = newCol.length - 1; row > 0; row--) {
+          if (newCol[row] === newCol[row - 1]) {
+            newCol[row] *= 2;
+            this.score += newCol[row];
+            newCol[row - 1] = 0;
+          }
+        }
+        newCol = newCol.filter((val) => val !== 0);
+
+        while (newCol.length < 4) {
+          newCol.unshift(0);
+        }
+
+        for (let row = 0; row < 4; row++) {
+          this.state[row][col] = newCol[row];
+        }
+      }
+      this.getRandomCell();
+      this.checkStatus();
```

Comment:
move this logic from all "arrows" function to this function
<img width="424" alt="image" src="https://github.com/user-attachments/assets/e1d51904-bf87-4eee-be01-5bc71bcba5c3">
it will simplify the logic